docs/pages/build/android-builds.md

```diff
@@ -0,0 +1,125 @@
+
+Your application's keystore should be kept private. **Under no circumstances should you check it in to your repository.** Debug keystores are the only exception because we don't use them for uploading apps to the Google Play store.
+
+### Configuring Gradle
+
+Let's focus on building a release app binary. Like we previously mentioned, your app binary needs to be signed with the keystore. Because we're building the project on a remote server we had to come up with a way of providing Gradle with the credentials which are not checked in to the repository. When running `expo eas:build --platform android`, we're writing the `android/app/eas-build.gradle` file with the following contents:
+
+```groovy
+android {
+  signingConfigs {
+    release {
+      // This is necessary to avoid needing the user to define a release signing config manually
+      // If no release config is defined, and this is not present, build for assembleRelease will crash
+    }
+  }
+
+  buildTypes {
+    release {
+      // This is necessary to avoid needing the user to define a release build type manually
+    }
+  }
+}
+
+project.afterEvaluate {
+  android.signingConfigs.release { config ->
+    def debug = gradle.startParameter.taskNames.any { it.toLowerCase().contains('debug') }
+
+    if (debug) {
+      return
+    }
+
+    def credentialsJson = rootProject.file("../credentials.json");
+
+    if (credentialsJson.exists()) {
+      if (config.storeFile) {
+        println("Path to release keystore file is already set, ignoring 'credentials.json'")
+      } else {
+        try {
+          def credentials = new groovy.json.JsonSlurper().parse(credentialsJson)
+
+          storeFile rootProject.file("../" + credentials.android.keystore.keystorePath)
+          storePassword credentials.android.keystore.keystorePassword
+          keyAlias credentials.android.keystore.keyAlias
+          keyPassword credentials.android.keystore.keyPassword
+        } catch (Exception e) {
+          println("An error occurred while parsing 'credentials.json': " + e.message)
+        }
+      }
+    } else {
+      if (config.storeFile == null) {
+        println("Couldn't find a 'credentials.json' file, skipping release keystore configuration")
+      }
+    }
+  }
+
+  android.buildTypes.release { config ->
+    config.signingConfig android.signingConfigs.release
+  }
+}
+```
+
+The most important part is the `release` signing config. It's configured to read the keystore and passwords from the `credentials.json` file at the project root. Even though you're not required to create this file on your own, it's created and populated with your credentials by EAS Builds before running the build. Please note that if `android.signingConfigs.release.storeFile` is already specified in your `build.gradle` we will **not** override your configuration.
```

Comment:
we could use the `@info` markup ([like this](https://github.com/expo/expo/blame/master/docs/pages/tutorial/image.md#L24)/[see that live](https://docs.expo.io/tutorial/image/#displaying-the-image-in-the-app)) to add yellow highlights on the parts that are relevant to make it easier to spot

Reply:
🤯

Reply:
I'll try to use it in a separate PR.